### PR TITLE
Improvements to the import modal

### LINKED ISSUE #127: Cannot import large contact lists
If you try importing a large contact list (not sure about the exact required minimum, I tested with 1000 contacts), the import process will always fail with the following error:

import large contact list

**Steps to reproduce:**

1. Download sample file
2. Import it into the contacts app
3. Wait for the import process to reach 100%
4. Notice the console error

**Sample file:**

Contact me

### LINKED ISSUE #137: Allow for canceling import process
Currently the import process cannot be cancelled. Only the export modal can be cancelled.

Originally I thought that it wouldn't make a lot of sense to cancel the import operation in the middle of the process because, unlike export, cancelling it leaves you with partially imported contacts in your contact list. Cancelling export has instead no effect on the contact list.

However, this possibility should probably be offered to the user because otherwise one can be stuck waiting a long time in case of importing long lists of contacts.

### LINKED ISSUE #140: Displaying errors for import process
Currently, if a user imports a list of contacts, after the import process we display how many of them have been succesfully imported, but do not specify why the others failed. However in v3 such information is displayed (specifically, the error given by the API is displayed).

We could display a generic error message specifying the contact that failed and the stage at which it failed (parsing, encrypting, or saving to database). I.e.
```
The 10th contact in your list could not be imported. Failed to parse it.
The 21st contact in your list could not be imported. Failed to encrypt it. (hopefully this should never happen).
The 29th contact in your list could not be imported. -- display API error --
```

### PR DESCRIPTION
**Pull request by econdepe**
- [x] Split import in batches to avoid timeout problems (fixes #127) 
- [x] Display information on errors during import (closes #140)
- [x] Load keys before opening import modal.
- [x] Generically improve code.
- [x] Possibility of cancelling import process (closes #137)
- [x] Display customized error for vcards version < 4.0 
- [x] Reserve 5% progress slots for import and parsing

### CODE REVIEW

File: src/app/components/import/ImportingModalContent.js [hunk 1]
```diff
@@ -1,115 +1,221 @@
...
     const api = useApi();
 
-    const [step, setStep] = useState(PARSING);
-    const [track, setTrack] = useState({
-        total: 0,
+    const [loading, withLoading] = useLoading(true);
+    const [model, setModel] = useState({
+        total: vcardContacts.length,
+        parsed: vcardContacts.map((contact, index) => {
+            index, contact;
```
Comment (mmso) [inline src/app/components/import/ImportingModalContent.js]:
```suggestion
            index, contact
```

File: src/app/components/import/ImportingModalContent.js [hunk 2]
```diff
@@ -1,115 +1,221 @@
...
-            !loadingKeys && encryptContacts();
-        }
-        if (step === IMPORTING) {
-            importContacts(track.encrypted);
+            for (let i = 0; i < apiCalls; i++) {
+                // avoid overloading API in the unlikely case submitBatch is too fast
+                await Promise.all([
+                    submitBatch({ contacts: contactBatches[i], labels }, { signal }),
+                    wait(API_SAFE_INTERVAL)
+                ]);
```
Comment (mmso) [inline src/app/components/import/ImportingModalContent.js]: Could perhaps be good with a `signal.aborted && break;`?

File: src/app/components/import/ImportingModalContent.js [hunk 3]
```diff
@@ -1,115 +1,221 @@
...
+                    if (Code === SUCCESS_IMPORT_CODE) {
+                        acc.submittedBatch.push(ID);
+                    } else {
+                        acc.failedOnSubmitBatch.push({ index, message: createSubmitErrorMessage(index, Error) });
+                    }
+                    return acc;
+                },
+                { submittedBatch: [], failedOnSubmitBatch: [] }
+            );
+            !signal.aborted &&
```
Comment (mmso) [inline src/app/components/import/ImportingModalContent.js]: perhaps more clear with `if (signal.aborted) { return; }` above `const { submittedBatch, failedOnSubmitBatch } = responses.reduce(`?

Comment (econdepe) [reply src/app/components/import/ImportingModalContent.js]: I wonder if it's possible (although clearly highly unlikely) that `signal.aborted` becomes `true` during the execution of the `reduce`. In such a case, `setModel` could be called after the component is unmounted, triggering a React complain

Comment (mmso) [reply src/app/components/import/ImportingModalContent.js]: Shouldn't be possible since it's a sync op

**Pull request merged**